messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+                null));
+        DeadLetterJpaConverter<M> converter = (DeadLetterJpaConverter<M>) converters
+                .stream()
+                .filter(c -> c.canConvert(entry.getMessage()))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("No such converter"));
+        return new JpaDeadLetter<>(entry,
+                                   deserializedDiagnostics,
+                                   converter.fromEntry(entry.getMessage(), serializer));
+    }
+
+    @Override
+    public boolean isFull(@Nonnull Object sequenceIdentifier) {
+        String identifier = toIdentifier(sequenceIdentifier);
+        return maximumNumberOfQueuesReached(identifier) || maximumQueueSizeReached(identifier);
+    }
+
+    @Override
+    public long maxSequences() {
+        return maxSequences;
+    }
+
+    @Override
+    public long maxSequenceSize() {
+        return maxSequenceSize;
+    }
+
+    @Override
+    public boolean process(@Nonnull Predicate<DeadLetter<? extends M>> sequenceFilter,
+                           @Nonnull Function<DeadLetter<? extends M>, EnqueueDecision<M>> processingTask) {
+
+        Optional<JpaDeadLetter<M>> first = findFirstLetterOfEachAvailableSequence()
+                .stream()
+                .map(this::toLetter)
+                .filter(sequenceFilter)
+                .findFirst();
+        if (!first.isPresent()) {
+            return false;
+        }
+
+        JpaDeadLetter<M> deadLetter = first.get();
+        claimDeadLetter(deadLetter);
+
+        while (deadLetter != null) {
+            EnqueueDecision<M> decision = processingTask.apply(deadLetter);
+            if (!decision.shouldEnqueue()) {
+                JpaDeadLetter<M> oldLetter = deadLetter;
+                DeadLetterEntry deadLetterEntry = findNextDeadLetter(oldLetter);
+                if (deadLetterEntry != null) {
+                    deadLetter = toLetter(deadLetterEntry);
+                    claimDeadLetter(deadLetter);
+                } else {
+                    deadLetter = null;
+                }
+                evict(oldLetter);
+            } else {
+                requeue(deadLetter,
+                        l -> decision.withDiagnostics(l)
+                                     .withCause(decision.enqueueCause().orElse(null))
+                                     .markTouched()
+                );
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Fetches the first letter for each sequence in the provided {@code processingGroup}. This is th message with the
+     * lowest index.
+     * <p>
+     * This fetches only letters which are available, having a {@code processingStarted} of null or longer ago than the
+     * configured {@code claimDuration}.
+     *
+     * @return A list of first letters of each sequence
+     */
+    private List<DeadLetterEntry> findFirstLetterOfEachAvailableSequence() {
```

Comment:
Perhaps this is a future adjustment we can make, but it's good to keep in mind that there might be a lot of available sequences. 
Doing a form of batching might be smart to do (in the future).